store.cpp

```diff
@@ -1,8 +1,168 @@
+bool bought[3];
+const int potion_price = 5;
+int actual_potion_price;
+const string potion_text = "Poção (Recupera X de vida)";
+const string store_welcome_text = "BEM VINDO MEU PARCEIRO\n";
+const string already_bought_text = "Você já comprou esse item, escolha novamente\n";
+const string exit_text = "Xau xau meu parceiro\n";
+const string not_enought_coins_text = "Sinto muito, você não me parece ter dinheiro suficiente para comprar isso\n";
+const string invalid_option_text= "Opção Inválida, escolha novamente\n";
+
+void see_store(main_character character) {
+    calculate_actual_potion_price(character);
+    printf("%s",store_welcome_text );
+    int game_progress_multiplyer = character.game_progress_multiplyer;
+    reset_is_bought();
+    itens[0] = generate_weapon(game_progress_multiplyer);
+    itens[1] = generate_armor(game_progress_multiplyer);
+    itens[2] = generate_rand_item(game_progress_multiplyer);
+    run_store(character,game_progress_multiplyer);
+    
+};
+
+void calculate_actual_potion_price(main_character main_character) {
+    actual_potion_price = potion_price * character.game_progress_multiplyer;
+}
+
+void run_store(main_character character,int game_progress_multiplyer) {
+    printf("\n\n");
+    print_menu(character,game_progress_multiplyer);
+    buying_menu(character);
+}
+
+void reset_is_bought(){
+    for (int i = 0; i< 3; i ++) {
+        bought[i] = false;
+    }
+};
+
+void exit_store() {
+    printf("%s",exit_text);
+};
+
+void buy_item(main_character character, item item,int index) {
+    if (can_buy_it(character,item.price)) {
+        character.coins -= item.price;
+        equip_item(character,item,index);
+        bought[index] = true;
+    } else {
+        printf("%s",not_enought_coins_text);
+    }
+    run_store(character,character.game_progress_multiplyer);
+};
+
+void equip_item(main_character character, item item, int index) {
+    item old_item;
+    if (item.type == ARMOR_TYPE) {
+        old_item =  character.armor;
+        character.armor = item;
+    } else if (item.type == WEAPON_TYPE){
+        old_item =  character.weapon;
+        character.weapon = item;
+    }
+    int def_diff = item.defense - old_item.defense;
+    int str_diff = item.strength - old_item.strength;
+    int hp_diff = item.health - old_item.health;
+
+    character.strength += str_dif;
+    character.defense += def_diff;
+    character.max_health += hp_diff;
+
+    if (character.max_health < character.health) {
+        character.health = character.max_health;
+    }
+}
+
+
+void buy_potion(main_character character) {
+    if (can_buy_it(character,potion_price)) {
+        character.coins -= actual_potion_price;
+        character.potion += 1;
+    } else {
+        printf("%s",not_enought_coins_text);
+    }
+    run_store();
+};
+
+void buying_menu(main_character character) {
+     int option;
+     while (true) {
+        scanf("%d",&option);
+        if (is_bought(option)) {
+            if (option == 1) {
```

Comment:
don't you think that a switch case would be better in this case? I guess it would be much more readable

Reply:
nop, dont like then